wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/probe/ProbeScheduler.java

```diff
@@ -216,6 +256,49 @@ public void cancel() {
     }
   }
 
+  private class DelayedSchedulingTask implements Runnable {
+    private final String workspaceId;
+    private final Supplier<WorkspaceStatus> statusSupplier;
+    private final WorkspaceProbes probes;
+    private final Consumer<ProbeResult> probeResultConsumer;
+
+    DelayedSchedulingTask(
+        Supplier<WorkspaceStatus> statusSupplier,
+        WorkspaceProbes probes,
+        Consumer<ProbeResult> probeResultConsumer) {
+      this.workspaceId = probes.getWorkspaceId();
+      this.statusSupplier = statusSupplier;
+      this.probes = probes;
+      this.probeResultConsumer = probeResultConsumer;
+    }
+
+    @Override
+    public void run() {
+      WorkspaceStatus status;
+
+      try {
+        status = statusSupplier.get();
+      } catch (RuntimeException e) {
+        // delay
+        return;
```

Comment:
The only possible way that I see is creating interface WorkspaceStatusSupplier with a defined checked exception. But it requires creation internal interface only for one method, Do you think it would be better?